.travis.yml

```diff
@@ -54,7 +54,7 @@ install:
   - spack install leveldb
   - spack install gotcha@0.0.2
   - spack install flatcc
-  - spack install margo^mercury~boostsys
+  - spack install margo^mercury+bmi~boostsys
```

Comment:
Is this just an option we need to document for users to do when building on a system with Infiniband? Or do we need to always require this? 
We could:
1. Just document it and let the user designate when to use it or not: `spack install margo^mercury+bmi`
2. Make it a variant in our spackage: 
    ```
    variant('infiniband', default='False', description="Enable InfiniBand support. Uses Mercury BMI plugin."
    ⋮
    depends_on('mercury+bmi', when='+infiniband')
    ```
3. Always require it in our spackage: `depends_on('mercury+bmi')`, which means we can't build without it when using spack

If the second option, do we use both libfabric and BMI at the same time, or is it a one-or-the-other option? If the latter, we could omit `ofi` from Mercury as well: `depends_on('mercury+bmi~ofi', when='+infiniband')`.

Reply:
Right now, the '+bmi' is required for server-server margo, regardless of the physical network type. I'm using 'bmi+tcp' as the mercury transport, which just uses TCP sockets for interprocess communication. So, if we try running with a mercury that doesn't support the bmi variant, server initialization will fail.

I didn't turn off the ofi variant since it only seems problematic on IBM SpectrumMPI systems like Summit/Sierra. We should document the problem for those systems and tell users to disable ofi. We need to test on other Infiniband clusters to see if it interferes with their MPI too.

Reply:
So it sounds like we'll want `depends_on('mercury+bmi')` in our spackage so users get a heads up before server initialization fails.

If that's the case, we could also put 
```
conflicts('^spectrum-mpi, when='^mercury+ofi' msg='Install with ^mercury~ofi or use a different mpi library')
```
in our spackage.

Reply:
Yes, I think we should depend on mercury+bmi in spack for now. I'm not sure if mercury+ofi conflicts with all SpectrumMPI systems, or just the infiniband ones like Summit/Sierra.

Reply:
Ah, okay. Thank you. I'll make an note about that but won't include a conflicts.

Lastly, we also conflict if the shared-memory plugin for Mercury is disabled (`mercury~sm`), correct?

Reply:
Yes, we always need +sm.